cassandra/components.py

```diff
@@ -1161,3 +1160,390 @@ def run_component(self):
+
+    This component is used instead of the HectorStubComponent to provide `tgav` to fldgen.
+
+    The component provides two capabilities:
+    :param Tgav:                        A data frame containing global mean temperature with fields
+                                        ['year', 'scenario', 'variable', 'value', 'units']; where value is 'Tgav'
+    :type Tgav:                         Pandas DataFrame
+
+    :param tgav_metadata:               A dictionary of metadata for the `Tgav` capability including:
+                                        [rds_file, scenario, climate_var_name, source_climate_data, units,
+                                        count, mean, median, min, max, std, na_count, null_count, all_finite]
+                                        designations
+    :type tgav_metadata:                dict
+
+
+    The parameters accepted by this component are:
+    :param rds_file:                    Full path with file name and extension to the input RDS emulator file
+                                        containing `tgav` outputs for each scenario
+    :type rds_file:                     str
+
+    :param climate_var_name:            Parent climate model variable name (e.g., tasAdjust) found in the file
+                                        name for the supporting climate data used by the emulator
+    :type climate_var_name:             str
+
+    :param scenario:                    Scenario name (e.g., rcp26)
+    :type scenario:                     str
+
+    :param units_field:                 Unit name from the `tgav` data
+    :type units_field:                  str
+
+    :param start_year:                  Start year of the climate data
+    :type start_year:                   int
+
+    :param through_year:                Year climate data goes through
+    :type through_year:                 int
+
+    """
+
+    # RDS file variable names
+    RDS_TGAV_NAME = 'tgav'
+    RDS_INFILES_NAME = 'infiles'
+
+    # capability name
+    TGAV_CAPABILITY_NAME = 'Tgav'
+
+    # output field order for the tgav data frame
+    TGAV_FIELD_ORDER = ['year', 'scenario', 'variable', 'value', 'units']
+
+    # component expected configuration fields
+    RDS_FILE_FIELD = 'rds_file'
+    CLIMATE_VAR_NAME_FIELD = 'climate_var_name'
+    SCENARIO_FIELD = 'scenario'
+    UNITS_FIELD = 'units'
+    START_YEAR_FIELD = 'start_year'
+    THROUGH_YEAR_FIELD = 'through_year'
+
+    def __init__(self, cap_tbl):
+        super(TgavStubComponent, self).__init__(cap_tbl)
+
+        # add capabilities to Cassandra
+        self.addcapability(self.TGAV_CAPABILITY_NAME)
+        self.addcapability('tgav_metadata')
+
+    def run_component(self):
+        """Run the TgavStubComponent component"""
+
+        # ensure required params are present
+        self.validate_params()
+
+        # convert the ListVector object from reading a RDS file to a Python dictionary
+        rds_dict = self.rds_to_dict()
```

Comment:
@crvernon, I was just discussing this with @FeralFlows, and it looks like we may be running into memory exhaustion.  If so, then those problems are surely exacerbated by storing the entire rds file contents in a member variable, since that ensures that the data can't be gc'd until the component goes out of scope, which is basically the whole run.  As far as I can tell, it doesn't look like we've fixed the memory bloat in the fldgen save files yet, so this adds up to a big chunk of memory.

I still think that having fldgen provide a function to extract the data we need is the right way to go, but at the very least, we need to manage the scope of these large objects very carefully.

Thinking on it a little more, I wonder if making this `TgavStub` a separate component is the right move at all.  The `FldgenComponent` already reads in this data anyhow.  What if we provided a configuration option to have `FldgenComponent` provide a `tgav` capability?  Then, if that capability is turned on, fldgen would extract the needed information from the files it is already reading in, use those values of `tgav` itself, and export them to the rest of the framework.  If not, then it skips that step and requests `tgav` from elsewhere in the framework.

A slight variant on that would be to implement the feature in issue #62.  Then, configuring fldgen to compute its own tgav would always trigger the export, but a user could squelch that if they wanted to have the rest of the system get tgav from somewhere else.  That would provide the most flexibility.